R/mean-model.R

```diff
@@ -0,0 +1,250 @@
+        pathsPerCompoundList[[cmp]]  <- getPathsForMoleculeAmount(simulation = loadedModel,cmp)
+        addOutputs(quantitiesOrPaths = pathsPerCompoundList[[cmp]], simulation = loadedModel) #add outputs for current compound to loadedModel's output list
+      }
+
+      simulationResults <- runSimulation(loadedModel)
+      simulationOutputs <- getOutputValuesTLF(simulationResults = simulationResults,
+                                              quantitiesOrPaths = simulationResults$allQuantityPaths,
+                                              individualIds = 0,
+                                              population =  loadPopulation(csvPopulationFile = "./data/popData.csv")
+      )  #using this placeholder population CSV file for now...need to make optional the population input to getOutputValuesTLF
+
+      self$massBalance <- list()
+
+      for ( cmp in compoundList ){ #For loop to calculate mass balance for each compound
+        self$massBalance[[cmp]] <- rep(0,nrow(simulationOutputs$data)) #Set up vector of zeros that is the same length as the number of simulation timepoints
+        pthsVec <- envList2PathStringsVector(pathsPerCompoundList[[cmp]]) #Get string vector of all outputs for the current compound
+        for (t in seq(1:nrow(simulationOutputs$data))){ #For each timepoint t
+          self$massBalance[[cmp]][t] <- sum(simulationOutputs$data[t,pthsVec]) # Select the columns of the dataframe self$simulationOutputs$data corresponding to the current compound and sum together the amounts of the compound in each path given in pthsVec for the timepoint t
+        }
+      }
+      self$massBalance <- as.data.frame(self$massBalance) #Convert massBalance list to datafram
+
+      # print(paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-outputs.csv"))
+      if (saveMassBalance){
+        write.csv(self$massBalance,file = paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-mass-balance.csv"))
+      }
+
+
+
+    },
+
+
+    calculateMeanModelPKParameters = function(userDefinedPKFunctions = NULL , savePKAnalysis = FALSE){
+
+      #userDefinedPKFunctions is an array of UserDefinedPKFunction objects
+
+      # Calculate MeanModel PK Parameters (details s. #32 )
+      # Sometimes it's required to calculate "non-standard" PK-Parameter which are not part of the basis
+      # toolbox. In such a case user MUST provide a function for calculation.
+
+      self$PKAnalysis <- calculatePKAnalyses(self$simulationResults)
+      tempFileName <- paste0("./tests/tempPKresults_",as.numeric(Sys.time()),".csv")
+      exportPKAnalysesToCSV(pkAnalyses = self$PKAnalysis , filePath = tempFileName)
+      PKResultsDF <- read.csv(file = tempFileName, check.names = FALSE)
+      file.remove(tempFileName)
+      #print(PKResultsDF)
+      #print(PKResultsDF[["IndividualId"]])
+      #print(colnames(PKResultsDF))
+
+
+      userDefinedPKResults = list()
```

Comment:
Any reason why you are not using the construct <- list(IndividualId<-NULL, Parameter<-null etc..)